debug/satellites-custom-layer.js

```diff
@@ -1,40 +1,13 @@
-const EARTH_RADIUS_METERS = 6371008.8;
-const EARTH_CIRCUMFERENCE_METERS = 2 * Math.PI * EARTH_RADIUS_METERS;
-const GLOBE_CIRCUMFERENCE_ECEF = 8192;
-const METERS_TO_ECEF = GLOBE_CIRCUMFERENCE_ECEF / EARTH_CIRCUMFERENCE_METERS;
-
 const KM_TO_M = 1000;
 const TIME_STEP = 3 * 1000;
 
-const globeVertCode = `
+const vertCode = `
     attribute vec3 a_pos_ecef;
     attribute vec3 a_pos_merc;
 
-    uniform mat4 u_projection;
-    uniform mat4 u_globeToMercMatrix;
-    uniform float u_globeToMercatorTransition;
-
-    void main() {
-        vec4 p = u_projection * u_globeToMercMatrix * vec4(a_pos_ecef, 1.);
-        p /= p.w;
-        if (u_globeToMercatorTransition > 0.) {
-            vec4 merc = u_projection * vec4(a_pos_merc, 1.);
-            merc /= merc.w;
-            p = mix(p, merc, u_globeToMercatorTransition);
-        }
-        gl_PointSize = 30.;
-        gl_Position = p;
-    }
-`;
-
-const mercVertCode = `
-    precision highp float;
-    attribute vec3 a_pos_merc;
-    uniform mat4 u_projection;
-
     void main() {
         gl_PointSize = 30.;
-        gl_Position = u_projection * vec4(a_pos_merc, 1.);
+        gl_Position = project_custom_layer(a_pos_merc, a_pos_ecef);
```

Comment:
We could simplify this one step further on the globe custom layer, since it's now consolidated in the same shader program the user may not need to provide two buffers (ecef + merc) and we could ask for them to only provide the merc while doing the projection to ecef for them in `project_custom_layer(a_pos_merc)`.

It would mean moving some computations from CPU to GPU but I think that's an ok tradeoff for the sake of simplified use. This also has the advantage that translating custom layer code to the new system becomes fairly trivial.

Reply:
Discussed offline: let's hold on this suggestion, e.g. transform ECEF on the GPU on our side for performance concern. We'll still keep API conservative and simple as in this PR and introduce other hooks as needed. The idea being that we don't over-commit by giving too much through the API, since it's always easy to give more later than give too much now and introduce breaking changes by removing it.